backend/pkg/schema/client.go

```diff
@@ -1,793 +1,390 @@
-		SetBody(&payload).
-		SetPathParam("subject", subject).
-		Put("/config/{subject}")
-	if err != nil {
-		return nil, fmt.Errorf("put config for subject failed: %w", err)
-	}
+// CompileProtoSchemaWithReferences compiles the given schema and its referenced schemas into protobuf files.
+// It recursively loads all schema references into an accessor map, which is then used by the protobuf compiler.
+func (c *CachedClient) CompileProtoSchemaWithReferences(
+	ctx context.Context,
+	schema sr.Schema,
+	accessorMap map[string]string,
+) (linker.Files, error) {
+	// Helper function to recursively fetch and parse all schema references.
+	var loadReferencesFn func(s sr.Schema) error
+	loadReferencesFn = func(s sr.Schema) error {
+		for _, ref := range s.References {
+			schemaRef, err := c.SchemaByVersion(ctx, ref.Subject, ref.Version)
+			if err != nil {
+				return err
+			}
+			accessorMap[ref.Name] = schemaRef.Schema.Schema
 
-	if res.IsError() {
-		restErr, ok := res.Error().(*RestError)
-		if !ok {
-			return nil, fmt.Errorf("put config for subject failed: Status code %d", res.StatusCode())
+			if err := loadReferencesFn(schemaRef.Schema); err != nil {
+				return fmt.Errorf("failed to parse reference schema %q: %w", ref.Subject, err)
+			}
 		}
-
-		return nil, restErr
-	}
-
-	parsed, ok := res.Result().(*PutConfigResponse)
-	if !ok {
-		return nil, fmt.Errorf("failed to parse config for subject response")
+		return nil
 	}
 
-	return parsed, nil
-}
-
-// DeleteSubjectConfig deletes compatibility level for a given subject.
-// If the subject you ask about does not have a subject-specific compatibility level set, this command returns an
-// error code.
-func (c *Client) DeleteSubjectConfig(ctx context.Context, subject string) (*ConfigResponse, error) {
-	res, err := c.client.R().
-		SetContext(ctx).
-		SetResult(&ConfigResponse{}).
-		SetPathParam("subject", subject).
-		Delete("/config/{subject}")
-	if err != nil {
-		return nil, fmt.Errorf("delete config for subject failed: %w", err)
+	// Load all references into accessorMap before proceeding.
+	if err := loadReferencesFn(schema); err != nil {
+		return nil, err
 	}
 
-	if res.IsError() {
-		restErr, ok := res.Error().(*RestError)
-		if !ok {
-			return nil, fmt.Errorf("delete config for subject failed: Status code %d", res.StatusCode())
-		}
+	// Add the main schema to the accessorMap with a temporary filename.
+	accessorMap[mainProtoFilename] = schema.Schema
 
-		return nil, restErr
+	// Compile the schema using the provided compiler.
+	sourceResolver := &protocompile.SourceResolver{
+		Accessor: protocompile.SourceAccessorFromMap(accessorMap),
 	}
-
-	parsed, ok := res.Result().(*ConfigResponse)
-	if !ok {
-		return nil, fmt.Errorf("failed to parse config for subject response")
+	compiler := protocompile.Compiler{
+		Resolver:       protocompile.WithStandardImports(c.standardImportsResolver(sourceResolver)),
+		SourceInfoMode: protocompile.SourceInfoStandard,
```

Comment:
Same concern here

Reply:
Changed to `protocompile.SourceInfoNone,`